event/subscriber.go

```diff
@@ -0,0 +1,77 @@
+package event
+
+import (
+	"context"
+	"fmt"
+	"github.com/medibloc/panacea-doracle/config"
+	log "github.com/sirupsen/logrus"
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+	"time"
+)
+
+type PanaceaSubscriber struct {
+	Client *rpchttp.HTTP
+}
+
+// NewSubscriber generates a rpc http client with websocket address.
+func NewSubscriber(conf *config.Config) (*PanaceaSubscriber, error) {
+	client, err := rpchttp.New(conf.Panacea.WSAddr, "/websocket")
+	if err != nil {
+		return nil, err
+	}
+
+	err = client.Start()
+	if err != nil {
+		return nil, err
+	}
+
+	return &PanaceaSubscriber{
+		Client: client,
+	}, nil
+}
+
+func (s *PanaceaSubscriber) Run(events ...Event) error {
+	log.Infof("start panacea event subscriber")
+
+	for _, e := range events {
+		query := e.GetEventType() + "." + e.GetEventAttributeKey() + "=" + e.GetEventAttributeValue()
+
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+		defer cancel()
+
+		fmt.Println(query)
+
+		txs, err := s.Client.Subscribe(ctx, "", query)
+		if err != nil {
+			return err
+		}
+
+		e := e
+		go func() {
+			for range txs {
+				t := <-txs
+				fmt.Println("got ", t.Events)
+				_ = e.EventHandler(t)
+			}
+		}()
+	}
+
+	return nil
+}
+
+//func convertEventTypeToEvent(eventType string) Event {
+//	switch eventType {
+//	case types.EventTypeRegistrationVote:
+//		return RegisterOracleEvent{}
+//	default:
+//		return nil
+//	}
+//}
+
+func (s *PanaceaSubscriber) Close() error {
+	log.Infof("closing panacea event subscriber")
+	if err := s.Client.Stop(); err != nil {
+		return err
+	}
+	return nil
```

Comment:
```suggestion
	return s.Client.Stop()
```
If you don't wrap the error from `s.Client.Stop()` or this `Close()` only call the `s.Client.Stop()`, you can just return it.

Reply:
Thanks for comment!